src/components/RegisterArtifactPopup/RegisterArtifactPopup.js

```diff
@@ -0,0 +1,191 @@
+import PropTypes from 'prop-types'
+import PopUpDialog from '../../common/PopUpDialog/PopUpDialog'
+import RegisterArtifactForm from '../../elements/RegisterArtifactForm/RegisterArtifactForm'
+import ErrorMessage from '../../common/ErrorMessage/ErrorMessage'
+
+import { v4 as uuidv4 } from 'uuid'
+
+import artifactApi from '../../api/artifacts-api'
+
+import './registerArtifactPopup.scss'
+
+const RegisterArtifactPopup = ({ match, refresh, setIsPopupDialogOpen }) => {
+  const [registerArtifactData, setRegisterArtifactData] = useState({
+    description: {
+      value: '',
+      required: false
+    },
+    kind: { value: 'general', required: false },
+    key: {
+      value: '',
+      required: false
+    },
+    target_path: {
+      value: '',
+      required: false
+    },
+    error: {
+      message: ''
+    }
+  })
+
+  const resetRegisterArtifactForm = useCallback(() => {
+    setRegisterArtifactData({
+      description: {
+        value: '',
+        required: false
+      },
+      kind: { value: 'general', required: false },
+      key: {
+        value: '',
+        required: false
+      },
+      target_path: {
+        value: '',
+        required: false
+      },
+      error: {
+        message: ''
+      }
+    })
+  }, [])
+
+  const registerArtifact = useCallback(() => {
+    if (
+      !registerArtifactData.key.value ||
+      !registerArtifactData.target_path.value
+    ) {
+      return setRegisterArtifactData(prevData => ({
+        ...prevData,
+        key: {
+          ...prevData.key,
+          required: !registerArtifactData.key.value
+        },
+        target_path: {
+          ...prevData.target_path,
+          required: !registerArtifactData.target_path.value
+        }
+      }))
+    }
+
+    if (registerArtifactData.error.message) {
+      setRegisterArtifactData(prevData => ({
+        ...prevData,
+        error: { ...prevData.error, message: '' }
+      }))
+    }
+
+    const uid = uuidv4()
+    const data = {
+      uid: uid,
+      key: registerArtifactData.key.value,
+      db_key: registerArtifactData.key.value,
+      tree: uid,
+      target_path: registerArtifactData.target_path.value,
+      description: registerArtifactData.description.value,
+      kind:
+        registerArtifactData.kind.value === 'general'
+          ? ''
+          : registerArtifactData.kind.value,
+      project: match.params.projectName,
+      producer: {
+        kind: 'api',
+        uri: window.location.host
+      }
+    }
+
+    if (registerArtifactData.kind.value === 'model') {
+      const {
+        target_path,
+        model_file
+      } = registerArtifactData.target_path.value.split('/').reduce(
+        (prev, curr, index, arr) => {
+          if (arr.length - 1 === index) {
+            prev.model_file = curr
+          } else {
+            prev.target_path += `${curr}/`
+          }
+          return prev
+        },
+        { model_file: '', target_path: '' }
+      )
```

Comment:
Here are two equivalent ways to get the same result:

Simple way:
```js
const parts = registerArtifactData.target_path.value.split('/')
const model_file = parts.pop()
const target_path = parts.join('/')
```
Advanced way:
```js
const { model_file, target_path } = registerArtifactData.target_path.value.match(/^(?:(?<target_path>.+\/))?(?<model_file>.+)$/)?.groups
```
I'm not saying you should change the code to use them, but just thought it might be nice to see :)